### DISCUSSION TITLE
Hosting without Cloud or Docker

### DISCUSSION THREAD

**Discussion by lrpatterson** [Budibase/budibase#3391] [Help] [answered]
I wanted to see if there was a way to host on a server without using docker or a cloud provider.  I am okay with doing the work to get it set up I just don't know where to start.

**Comment by mjashanks** [comment, marked as answer]
There is definitely a way, I don't think any of us has done it yet. I can give you some rough guidance though.

The `docker-compose.yaml` file is a great resource for seeing what's in the stack, and the default ports: https://github.com/Budibase/budibase/blob/develop/hosting/docker-compose.yaml

You could run each one of these components manually, by downloading and installing the non-docker installs from their website:
- MinIO - ` minio-service`
- CouchDB - `couchdb-service`
- Redis - `redis-service`
- Watchtower - `watchtower-service` (afaik this actually controls updates via docker, so I don't think it's necessary!)
- Envoy - `proxy-service`  - you will also need a config file for this `envoy.yaml` - https://github.com/Budibase/budibase/blob/develop/hosting/envoy.yaml
- Budibase App & Worker - `app-service` & `worker-service`, you can checkout the repo, and workout how these run by starting with the `package.json`: https://github.com/Budibase/budibase/blob/develop/package.json (see script `dev:server`)

Thats just a starting point for you - good luck :)

**Reply by lrpatterson** [reply]
Thank you so much for the reply.  I work at a stricter org and can't use docker or kub. to run stuff.

**Comment by daawaan4x** [comment]
Hello @lrpatterson, I'm currently also having the same requirements. Were you able to run Budibase without docker or k8s?
